pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1665,73 +1665,95 @@ private void parseQuery() {
     }
   }
 
-  private void updateRowBuffer() throws SQLException {
+  private void setRowBufferColumn(int columnIndex, Object valueObject) throws SQLException {
+    if (valueObject instanceof PGobject) {
+      String value = ((PGobject) valueObject).getValue();
+      rowBuffer.set(columnIndex, (value == null) ? null : connection.encodeString(value));
+    } else {
+      switch (getSQLType(columnIndex + 1)) {
+
+        // boolean needs to be formatted as t or f instead of true or false
+        case Types.BIT:
+        case Types.BOOLEAN:
+          rowBuffer.set(columnIndex, connection
+              .encodeString(((Boolean) valueObject).booleanValue() ? "t" : "f"));
+          break;
+        //
+        // toString() isn't enough for date and time types; we must format it correctly
+        // or we won't be able to re-parse it.
+        //
+        case Types.DATE:
+          rowBuffer.set(columnIndex, connection
+              .encodeString(
+                  connection.getTimestampUtils().toString(
+                      getDefaultCalendar(), (Date) valueObject)));
+          break;
+
+        case Types.TIME:
+          rowBuffer.set(columnIndex, connection
+              .encodeString(
+                  connection.getTimestampUtils().toString(
+                      getDefaultCalendar(), (Time) valueObject)));
+          break;
+
+        case Types.TIMESTAMP:
+          rowBuffer.set(columnIndex, connection.encodeString(
+              connection.getTimestampUtils().toString(
+                  getDefaultCalendar(), (Timestamp) valueObject)));
+          break;
+
+        case Types.NULL:
+          // Should never happen?
+          break;
+
+        case Types.BINARY:
+        case Types.LONGVARBINARY:
+        case Types.VARBINARY:
+          if (isBinary(columnIndex + 1)) {
+            rowBuffer.set(columnIndex, (byte[]) valueObject);
+          } else {
+            try {
+              rowBuffer.set(columnIndex,
+                  PGbytea.toPGString((byte[]) valueObject).getBytes("ISO-8859-1"));
```

Comment:
Why ISO-8859-1 ? The driver generally assumes UTF-8

Reply:
While a valid question, I do not have the answer. This i just pulled the loop out of this method, reducing the indentation level by one.

Reply:
hmmm ok, for now leave it but can you add a new issue to review this later